tests/unit/contentproviders/test_git.py

```diff
@@ -18,6 +19,28 @@ def test_clone(repo_with_content):
         assert git_content.content_id == sha1[:7]
 
 
+def test_submodule_clone(repo_with_submodule):
+    """Test git clone containing a git submodule."""
+    upstream, sha1_upstream, sha1_submod = repo_with_submodule
+
+    with TemporaryDirectory() as clone_dir:
+        submod_dir = os.path.join(clone_dir, 'submod')  # set by fixture
+        spec = {'repo': upstream}
+        git_content = Git()
+        for _ in git_content.fetch(spec, clone_dir):
+            pass
+        assert os.path.exists(os.path.join(clone_dir, 'test'))
+        assert os.path.exists(os.path.join(submod_dir, 'test'))
+
+        # get current sha1 of submodule
+        cmd = ['git', 'rev-parse', 'HEAD']
+        sha1 = subprocess.Popen(cmd, stdout=subprocess.PIPE, cwd=submod_dir)
+        submod_sha1 = sha1.stdout.read().decode().strip()
+
+        assert git_content.content_id == sha1_upstream[:7]
+        assert submod_sha1[:7] == submod_sha1[:7]
```

Comment:
For my education: what are we testing here or is it a typo? Maybe it should be `sha1_submod[:7] == submod_sha1[:7]`?

The only time you want to compare `a` to `a` that I know is when you want to find out if the contents is NaN, which hopefully we don't have here :)

Again for my education: why do we need the `[:7]` bit? Both SHAs are the result of calling the same `git rev-parse...` so they should both be of the same length no?

Reply:
Yes, I was just doing it for consistency to match `git_content.content_id` which was not created by me. To be clear, I am testing that the git submodule is at the commit it is supposed to be at. Since there is only one commit this is unlikely to change, but I felt like only testing that the 'test' file existed wasn't good enough.

And yeah, you're right I need to fix the variable names.